### PR TITLE
PtRef: Display line section impact in lines

### PR DESCRIPTION
**Pull request by xlqian**
https://jira.canaltp.fr/browse/NAVITIAII-2211

### CODE REVIEW

File: source/type/pb_converter.h [hunk 1]
```diff
@@ -322,6 +322,13 @@ struct PbCreator {
                 fill_pb_object(nav_obj, pb_list->Add());
             }
         }
+        template<typename Nav, typename Pb>
+        void fill_line_with_line_seciton_message(const std::vector<Nav*>& nav_list,
```
Comment (antoine-de) [inline source/type/pb_converter.h]: typo on seciton

File: source/type/pb_converter.cpp [hunk 2]
```diff
@@ -624,6 +624,35 @@ void PbCreator::Filler::fill_pb_object(const nt::Line* l, pbnavitia::Line* line)
     }
 
 }
+void PbCreator::Filler::fill_line_with_line_section_message(const nt::Line* l , pbnavitia::Line* line){
```
Comment (antoine-de) [inline source/type/pb_converter.cpp]: wouldn't it be better to have a `Filler` flag (like DumpMessage) activated only with `PbCreator::pb_fill` (but for all objects) ?

**Review by TeXitoi — APPROVED** [review]

**Review by antoine-de — COMMENTED** [review]
erf it's way more verbose then :disappointed: 

what do the others think about it ?

even if it's a bool, I prefer a bit an enum for sutff like this to 'name' the parameter (like DumpMessage)

File: source/type/pb_converter.cpp [hunk 3]
```diff
@@ -623,37 +623,34 @@ void PbCreator::Filler::fill_pb_object(const nt::Line* l, pbnavitia::Line* line)
...
+                    continue;
+                for (const auto& entity: impact_ptr->informed_entities()) {
+                    if (boost::get<nt::disruption::LineSection>(&entity) &&
+                            added_impact.insert(impact_ptr).second ) {
+                        fill_messages(vj.meta_vj, line);
+                        return true;
+                    }
                 }
             }
+            return true;
```
Comment (pbougue) [inline source/type/pb_converter.cpp]: shouldn't we return `false` here?

File: source/type/pb_converter.cpp [hunk 4]
```diff
@@ -623,37 +623,34 @@ void PbCreator::Filler::fill_pb_object(const nt::Line* l, pbnavitia::Line* line)
...
-                    fill_messages(vj.meta_vj, line);
-                    return true;
+    if (dump_message == DumpMessage::Yes && dump_line_section_message) {
+        /*
+         * Here we dump the impacts which impact LineSection.
+         * We could have link the LineSection impact with the line, but that would change the code and
+         * the behavior too much.
+         * */
+        std::set<boost::shared_ptr<nt::disruption::Impact>> added_impact;
+        auto pred = [&](const nt::VehicleJourney& vj) {
```
Comment (pbougue) [inline source/type/pb_converter.cpp]: `pred` is a bit too generic name (predicate, right?)

**Review by pbougue — COMMENTED** [review]
As for the too verbose (and typing the boolean), a solution might be to wrap it in a struct with `dump_message` (as both are defaulted, it would probably do the job)

File: source/type/pb_converter.cpp [hunk 5]
```diff
@@ -623,6 +623,32 @@ void PbCreator::Filler::fill_pb_object(const nt::Line* l, pbnavitia::Line* line)
...
+            for(const auto& impact: vj.meta_vj->impacted_by) {
+                auto impact_ptr = impact.lock();
+                if (! impact_ptr)
+                    continue;
+                for (const auto& entity: impact_ptr->informed_entities()) {
+                    if (boost::get<nt::disruption::LineSection>(&entity) &&
+                            added_impact.insert(impact_ptr).second ) {
+                        fill_messages(vj.meta_vj, line);
+                        return true;
```
Comment (pbougue) [inline source/type/pb_converter.cpp:648]: isn't it too early to return? if there are multiple line_sections?

Comment (xlqian) [reply source/type/pb_converter.cpp:648]: we can have multiple line_sections on the impact, but we add only once the impact :)

File: source/type/pb_converter.cpp (hunk 3 above)
Comment (xlqian) [reply source/type/pb_converter.cpp]: `false` will stop the loop, which is not wanted...

**Review by antoine-de — APPROVED** [review]

**Pull request merged**